Add sound tab and solve null registry keys preventing launch

General-101 opened the pull request.

Commit:
Add sound tab

Also begin work on repairing the registry if we fail to find a key

Commit:
Add prompt to select tool location

Also set keys for 64 and 32 bit

Commit:
Update key names

Commit:
Check if all keys exist before asking for H2Tool path

Commit:
Remove 64 bit key

It's not needed

num0005 commented on Launcher/MainWindow.xaml.cs:
```diff
@@ -215,7 +217,66 @@ public MainWindow()
+                    H2Ek_install_path_user_set = new FileInfo(H2Tool_path).Directory.FullName;
+                }
+
+                if (H2EK_Install_Path_key == null)
```
```suggestion
                if (H2EK_Install_Path_key is null)
```

num0005 commented on Launcher/MainWindow.xaml.cs:
```diff
@@ -215,7 +217,66 @@ public MainWindow()
+                        .CreateSubKey("SOFTWARE\\Microsoft\\Microsoft Games\\Halo 2\\1.0", true);
+                    registryKey32.SetValue("ToolsInstallDir", H2Ek_install_path_user_set + "\\");
+                }
+                if (Guerilla_Tag_key == null)
```
```suggestion
                if (Guerilla_Tag_key is null)
```

num0005 commented on Launcher/MainWindow.xaml.cs:
```diff
@@ -215,7 +217,66 @@ public MainWindow()
+                    .CreateSubKey("SOFTWARE\\Microsoft\\Microsoft Games\\Halo 2\\1.0", true);
+
+                bool H2EK_key = true;
+                bool Guerilla_key = true;
```
```C#
string install_path = null;
if (H2EK_Install_Path_key)
   install_path = GetPathFromH2EK(H2EK_Install_Path_key);
if (Guerilla_Tag_key )
   install_path = GetPathFromGuerilla(Guerilla_Tag_key );
if (install_path is null)
   // code to prompt for path
```

Commit:
This should be proper

Commit:
Changes to how we check keys

Commit:
Don't check if we can write to keys if they aren't null

Commit:
Changes to how we set path

Made H2Tool_Path local to function?

Merge H2Ek_install_path_user_set and H2Ek_install_path

If the user cancels H2tool dialog box then default to launcher path and warn them. Do not assign registry keys using the launcher path here because of what that could cause.

Commit:
Make registry edits a setting

Commit:
Undo that to make sure exes can be patched no matter what

Commit:
Keys being weird?
